### PR TITLE
chore(newrelic_nrql_alert_condition): move read to create function to avoid root resource not found

### PR DESCRIPTION
**Pull request by kidk**
# Description

Change to read function to avoid setId("") happening. I think we need to verify the resource has been created by reading it back before setting the Id.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

### CODE REVIEW

File: newrelic/resource_newrelic_nrql_alert_condition.go [hunk 1]
```diff
@@ -412,9 +413,25 @@ func resourceNewRelicNrqlAlertConditionCreate(ctx context.Context, d *schema.Res
 		return diag.FromErr(err)
 	}
 
-	d.SetId(serializeIDs([]int{d.Get("policy_id").(int), conditionID}))
+	retryErr := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_nrql_alert_condition.go:416]: Should we define a value for `schema.TimeoutCreate`? I wasn't sure if there was a default value.

File: newrelic/resource_newrelic_alert_policy_channel.go [hunk 2]
```diff
@@ -116,9 +118,26 @@ func resourceNewRelicAlertPolicyChannelCreate(ctx context.Context, d *schema.Res
 		return diag.FromErr(err)
 	}
 
+	retryErr := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_alert_policy_channel.go:121]: Should we define a value for schema.TimeoutCreate? I wasn't sure if there was a default value.

File: newrelic/resource_newrelic_nrql_alert_condition.go (hunk 1 above)
Comment (kidk) [reply newrelic/resource_newrelic_nrql_alert_condition.go:420]: I think it has a default, but I added it to be sure. Thanks 👍

File: newrelic/resource_newrelic_alert_policy_channel.go (hunk 2 above)
Comment (kidk) [reply newrelic/resource_newrelic_alert_policy_channel.go:125]: I think it has a default, but I added it to be sure. Thanks 👍

**Review by mbazhlekova — APPROVED** [review]

**Pull request merged**